src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private File file;
+    private final StringBuilder builder = new StringBuilder();
+    private int buy = 0;
+    private int supply = 0;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        file = new File(fromFileName);
+
+        readFromFile();
+        writeToFile(toFileName);
+    }
+
+    public void readFromFile() {
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String value;
+            while ((value = reader.readLine()) != null) {
+                builder.append(value);
+                builder.append("\n");
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("file read error " + e);
+        }
+        String[] array = builder.toString().split("\n");
+
+        for (String s : array) {
+            String[] data = s.split(",");
+            this.buy += data[0].equals("buy") ? Integer.parseInt(data[1]) : 0;
+            this.supply += data[0].equals("supply") ? Integer.parseInt(data[1]) : 0;
+        }
```

Comment:
The method with the name readFromFile must only READ FROM FILE. Move other logic to the separate private method. Your code must be reusable 
Let's move logic related to the file writing to the separate private method